src/main/java/com/thealgorithms/strings/Anagrams.java

```diff
@@ -0,0 +1,150 @@
+/** Author : Siddhant Swarup Mallick
+ * Github : https://github.com/siddhant2002
+ */
+
+/** PROBLEM DESCRIPTION :
+ * An anagram is a word or phrase formed by rearranging the letters of a different word or phrase, typically using all the original letters exactly once.[1] For example, the word anagram itself can be rearranged into nag a ram, also the word binary into brainy and the word adobe into abode. Reference from https://en.wikipedia.org/wiki/Anagram
+ */
+
+package com.thealgorithms.strings;
+import java.util.*;
+public class Anagrams 
+{
+    // 4 approaches are provided for anagram checking. approach 2 and approach 3 are similar but differ in running time.
+    public static void main(String args[]) {
+        String first = "deal";
+        String second = "lead";
+        // All the below methods takes input but doesn't return any output to the main method.
+        Anagrams nm=new Anagrams();
+        nm.approach2(first, second);  /* To activate methods for different approaches*/
+        nm.approach1(first, second);  /* To activate methods for different approaches*/
+        nm.approach3(first, second);  /* To activate methods for different approaches*/
+        nm.approach4(first, second);  /* To activate methods for different approaches*/
+
+        /**
+         * OUTPUT :
+         * first string ="deal" second string ="lead"
+         * Output: Anagram
+         * Input and output is constant for all four approaches
+         * 1st approach Time Complexity : O(n logn)
+         * Auxiliary Space Complexity : O(1)
+         * 2nd approach Time Complexity : O(n)
+         * Auxiliary Space Complexity : O(1)
+         * 3rd approach Time Complexity : O(n)
+         * Auxiliary Space Complexity : O(1)
+         * 4th approach Time Complexity : O(n)
+         * Auxiliary Space Complexity : O(n)
+         */
+    }
+
+    void approach1(String s, String t) 
```

Comment:
Please return a boolean value from the functions instead of printing to console.

Reply:
ok. Thanks